gamemode/core/hooks/sv_hooks.lua

```diff
@@ -722,7 +722,8 @@ function GM:CreateSalaryTimer(client)
 	if (not character) then return end
 
 	local faction = nut.faction.indices[character:getFaction()]
-	if (not faction or not isnumber(faction.pay) or faction.pay <= 0) then
+	local class = nut.class.indices[character:getClass()]
+	if ((not faction or not isnumber(faction.pay) or faction.pay <= 0) and (not class or not isnumber(class.pay) or class.pay <= 0)) then
```

Comment:
Wouldn't this return if the character is not part of a class or the class in question doesn't have a custom pay?
By default, characters are not assigned to a class. That will pass in the "not class" and simply return.
If the character is part of a class, but its pay is not defined (for instance the class is supposed to use the default faction pay), it will pass the "not isnumber(class.pay) or class.pay <= 0)" and also return